### PR TITLE
itest: make harnesses usable in external projects

### PR DESCRIPTION
**Pull request by guggero**
To use the node and network harnesses from the lntest package in other projects, it is necessary to fix the macaroon path to have all macaroons in the same directory (without this fix, the admin, readonly and invoice macaroon are in the `data/` directory of lnd while all the sub-server macaroons are in the `data/chain/bitcoin/regtest/` folder).
Also it is useful if we can specify the location of the `lnd-itest` binary so we don't have to use the same folder structure/depth in other projects.
Finally, to get access to a node's data directory, we need to export the node's config struct.

### CODE REVIEW

File: lntest/node.go [hunk 1]
```diff
@@ -232,7 +232,7 @@ func (cfg nodeConfig) genArgs() []string {
 // harness. Each HarnessNode instance also fully embeds an RPC client in
 // order to pragmatically drive the node.
 type HarnessNode struct {
-	cfg *nodeConfig
+	Cfg *nodeConfig
```
Comment (wpaulino) [inline lntest/node.go]: Should also export `nodeConfig`.

File: lntest/node.go [hunk 2]
```diff
@@ -298,9 +298,13 @@ func newNode(cfg nodeConfig) (*HarnessNode, error) {
...
 	cfg.TLSCertPath = filepath.Join(cfg.DataDir, "tls.cert")
 	cfg.TLSKeyPath = filepath.Join(cfg.DataDir, "tls.key")
-	cfg.AdminMacPath = filepath.Join(cfg.DataDir, "admin.macaroon")
-	cfg.ReadMacPath = filepath.Join(cfg.DataDir, "readonly.macaroon")
-	cfg.InvoiceMacPath = filepath.Join(cfg.DataDir, "invoice.macaroon")
+
+	networkDir := filepath.Join(
+		cfg.DataDir, "chain", "bitcoin", cfg.NetParams.Name,
+	)
+	cfg.AdminMacPath = filepath.Join(networkDir, "admin.macaroon")
```
Comment (wpaulino) [inline lntest/node.go]: I'd imagine these don't need to be specified since we'd want them in their default path?

File: lntest/node.go (hunk 1 above)
Comment (guggero) [reply lntest/node.go]: Done.

File: lntest/node.go (hunk 2 above)
Comment (guggero) [reply lntest/node.go]: They are used in different places within the harness and need to be set.

File: lntest/node.go [hunk 3]
```diff
@@ -298,9 +298,13 @@ func newNode(cfg NodeConfig) (*HarnessNode, error) {
 	cfg.LogDir = filepath.Join(cfg.BaseDir, "log")
 	cfg.TLSCertPath = filepath.Join(cfg.DataDir, "tls.cert")
 	cfg.TLSKeyPath = filepath.Join(cfg.DataDir, "tls.key")
-	cfg.AdminMacPath = filepath.Join(cfg.DataDir, "admin.macaroon")
-	cfg.ReadMacPath = filepath.Join(cfg.DataDir, "readonly.macaroon")
-	cfg.InvoiceMacPath = filepath.Join(cfg.DataDir, "invoice.macaroon")
+
+	networkDir := filepath.Join(
```
Comment (Roasbeef) [inline lntest/node.go]: The commit message should have some rationale w.r.t why this change was required.

Comment (guggero) [reply lntest/node.go]: Yes, I agree. I added some context to the commit message.
